server/node/src/main/java/org/kaaproject/kaa/server/admin/controller/KaaAdminController.java

```diff
@@ -2106,53 +2106,77 @@ public void editUserConfiguration(@RequestBody EndpointUserConfigurationDto endp
     }
 
     /**
-     * Provides security credentials, allowing the endpoint with the given
-     * credentials to interact with the application specified.
-     * */
-    @RequestMapping(value = "provisionCredentials", params = { "applicationId", "credentials" }, method = RequestMethod.POST)
+     * Provides security credentials, allowing an endpoint that uses them to
+     * interact with the specified application.
+     *
+     * @param applicationId The application ID to allow interaction with
+     * @param credentialsBody The security credentials to save
+     *
+     * @return The security credentials saved
+     *
+     * @throws KaaAdminServiceException - if an exception occures.
+     */
+    @RequestMapping(value = "provideCredentials", params = { "applicationId", "credentialsBody" }, method = RequestMethod.POST)
     @ResponseBody
-    public CredentialsDto provisionCredentials (
+    public CredentialsDto provideCredentials(
             @RequestParam String applicationId,
-            @RequestParam String credentials)
-            throws KaaAdminServiceException {
-        //TODO: implement
-        return null;
+            @RequestParam String credentialsBody)
+                    throws KaaAdminServiceException {
+        return this.kaaAdminService.provideCredentials(applicationId, credentialsBody);
     }
 
     /**
-     * Provides registration information, allowing the endpoint with the given
-     * credentials to use this information during registration process.
-     * */
-    @RequestMapping(value = "provisionRegistrationInfo", params = { "applicationId", "credentialsId", "serverSideEPProfileVersion",
-            "serverSideEPProfileBody" }, method = RequestMethod.POST)
+     * Binds credentials to the server-side endpoint profile specified.
+     *
+     * @param applicationId The application ID
+     * @param credentialsId The ID of the credentials to bind
+     * @param serverProfileVersion The server-side endpoint profile version
+     * @param serverProfileBody The server-side endpoint profile body
+     *
+     * @throws KaaAdminServiceException - if an exception occures.
+     */
+    @RequestMapping(value = "provideRegistration", params = { "applicationId", "credentialsId" }, method = RequestMethod.POST)
     @ResponseStatus(HttpStatus.OK)
-    public void provisionRegistrationInfo(
-            @RequestParam String applicationId, 
+    public void provideRegistration(
+            @RequestParam String applicationId,
             @RequestParam String credentialsId,
-            @RequestParam Integer serverSideEPProfileVersion, 
-            @RequestParam String serverSideEPProfileBody)
+            @RequestParam(required = false) Integer serverProfileVersion,
```

Comment:
This should not be an optional parameter, because otherwise there is nothing to provision